### DISCUSSION TITLE
How to use builtin function of database in sea-orm?

### DISCUSSION THREAD

**Discussion by ofzo** [SeaQL/sea-orm#2295] [Q&A] [answered]
exp: I want to gen a sql like
```sql
INSERT INTO users (uid, username, email, birthdate, is_active) VALUES (UUID(), 'xxxx', 'xxxx.xx@xxx.com', '2020-12-20', true);
```
the `UUID()` is builtin function in mysql, I want to use it directly;

What should I do?

**Comment by Expurple** [comment, marked as answer]
Builtin functions can be expressed using `sea_query::Expr::cust("UUID()")`.

But if you have a SeaORM entity and want to use it normally through an [ActiveModel](https://docs.rs/sea-orm/latest/sea_orm/entity/trait.ActiveModelTrait.html), I only know how to sneak an expression into an UPDATE:

```rust
users::Entity::update_many()
    .col_expr(users::Column::Uid, Expr::cust("UUID()"))
    .filter(...)
    .exec(&conn)
    .await?;
```

[Insert APIs](https://docs.rs/sea-orm/latest/sea_orm/index.html?search=insert) seem to only accept whole [ActiveModel](https://docs.rs/sea-orm/latest/sea_orm/entity/trait.ActiveModelTrait.html)s, where fields are [ActiveValue](https://docs.rs/sea-orm/latest/sea_orm/entity/enum.ActiveValue.html)s, which can only store Rust values and not arbitrary SQL expressions.

---

If your use case only involves the `UUID()` function, there are a few obvious workarounds:

- Set a `DEFAULT` in the DB and keep this column [NotSet](https://docs.rs/sea-orm/latest/sea_orm/entity/enum.ActiveValue.html#) when you insert the [ActiveModel](https://docs.rs/sea-orm/latest/sea_orm/entity/trait.ActiveModelTrait.html).
- Generate and set the UUID on the Rust side.
